elastic-mapping-updater/src/test/resources/template1.json

```diff
@@ -1,184 +1,194 @@
 {
-    "settings": {
-        "index": {
-            "number_of_shards": 5,
-            "number_of_replicas": 0,
-            "codec": "best_compression",
-            "refresh_interval": "1s",
-            "max_result_window": "1000000",
-            "merge.scheduler.max_thread_count": 1
-        },
-        "analysis": {
-            "char_filter": {
-                "html_strip_filter": {
-                    "type": "html_strip"
-                }
-            },
-            "filter": {
-                "million_token_limit": {
-                    "type": "limit",
-                    "max_token_count": 1000000
-                },
-                "period_break_filter": {
-                    "type": "pattern_capture",
-                    "preserve_original": false,
-                    "patterns": [
-                        "^([\\\\p{Nd},\\\\.'?]+)$|([^\\\\.']+)"
-                    ]
-                },
-                "max_token_length_filter": {
-                    "type": "length",
-                    "max": 10922
-                }
-            },
-            "analyzer": {
-                "general_analyzer": {
-                    "type": "custom",
-                    "filter": [
-                        "million_token_limit",
-                        "max_token_length_filter",
-                        "icu_normalizer"
-                    ],
-                    "char_filter": [],
-                    "tokenizer": "icu_tokenizer"
-                },
-                "path_analyzer": {
-                    "type": "custom",
-                    "tokenizer": "path_analyzer",
-                    "filter": [
-                        "max_token_length_filter",
-                        "icu_normalizer"
-                    ]
-                },
-                "version_analyzer": {
-                    "type": "pattern",
-                    "pattern": "\\\\s+"
-                },
-                "standard_max_term_100": {
-                    "type": "standard",
-                    "stopwords": "_english_",
-                    "max_token_length": 100
-                }
-            },
-            "normalizer": {
-                "keyword_normalizer": {
-                    "type": "custom",
-                    "filter": [
-                        "icu_normalizer"
-                    ]
-                }
+    "template": {
+        "settings": {
```

Comment:
In `IndexTemplateRequest`, `settings` and `mappings` are now nested inside `template`

Old Index Template: https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-templates-v1.html
New Index Template: https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-put-template.html

Reply:
@kusumaghoshdastidar I know we no longer require the default fields now in the existing templates by using  `ApiTypeHelper.DANGEROUS_disableRequiredPropertiesCheck(true);` but how will get around the issue of `settings` and `mappings` being nested in `template`

Reply:
`settings` and `mappings` within `template` happens only with the new `_index_template` API which we aren't using so we should be ok with the deserializing of the `TypeMapping`.